workshop/exercise-2.md

```diff
@@ -147,56 +131,67 @@ Esto se hace encadenando `.chatReference(conversationId)` y `.chatMemory(chatMem
 
 Antes de cambiar a la memoria persistente, primero vamos a verificar el comportamiento de la implementación en memoria que configuraste en la Parte 2.
 
-1. Asegúrate de que tu `ChatConfig` esté proveyendo el bean de `InMemoryChatMemory`. Si ya lo cambiaste, vuelve a ponerlo temporalmente.
-
-```java
-// En ChatConfig.java
-@Bean
-public ChatMemory chatMemory() {
-    return new InMemoryChatMemory();
-}
-```
-
-2. Ejecuta la aplicación.
-3.  Abre tu navegador y haz una pregunta inicial: `http://localhost:8080/ai/chat?message=Mi nombre es Fulano. ¿Puedes decirme qué es Spring AI?`.
-4. En la misma pestaña (misma sesión), pregunta: `http://localhost:8080/ai/chat?message=¿Recuerdas mi nombre?`
+1. Ejecuta la aplicación.
+2. Abre tu navegador y haz una pregunta inicial: `http://localhost:8080/ai/chat?message=Mi nombre es Fulano. ¿Puedes decirme qué es Spring AI?`.
+3. En la misma pestaña (misma sesión), pregunta: `http://localhost:8080/ai/chat?message=¿Cuál es mi nombre?, `
     - __Resultado esperado__: El asistente responderá correctamente "Fulano", demostrando que la memoria funciona dentro de la sesión de la aplicación.
 
-5. Reinicia tu aplicación Spring Boot.
-6. Una vez reiniciada, en la misma pestaña del navegador, vuelve a preguntar: `http://localhost:8080/ai/chat?message=¿Recuerdas mi nombre?`
+4. Reinicia tu aplicación Spring Boot.
+5. Una vez reiniciada, en la misma pestaña del navegador, vuelve a preguntar: `http://localhost:8080/ai/chat?message=¿Cuál es mi nombre?`
     - __Resultado esperado__: El asistente NO recordará tu nombre. Esto demuestra la naturaleza volátil de `InMemoryChatMemory`: la memoria se borra cuando la aplicación se detiene.
 
 
 
-
-
 ### Parte 3 - Implementando Memoria Persistente con PostgreSQL y JDBC
 
-La memoria en memoria es útil, pero se pierde al reiniciar. Para una aplicación real, necesitamos persistencia. Ya que tienes PostgreSQL configurado, vamos a utilizarlo para almacenar el historial de chat. Gracias a la autoconfiguración de Spring AI, este cambio es sorprendentemente fácil.
+Almacenar información en memoria es útil, pero se pierde al reiniciar. Para una aplicación real, necesitamos persistencia. `JdbcChatMemoryRepository` es una implementación que utiliza JDBC para guardar los mensajes en una base de datos relacional, ideal para aplicaciones que requieren que el historial de chat sobreviva a los reinicios.
 
 1. Verificar Dependencias
 
-En tu `pom.xml`, asegúrate que se incluya:
-
+En tu `pom.xml`, asegúrate de que las siguientes dependencias estén incluidas. La nueva dependencia clave aquí es  `spring-ai-starter-model-chat-memory-repository-jdbc`, que nos proporciona la autoconfiguración para la memoria de chat con JDBC.
+```xml
+<!-- Proporciona la autoconfiguración para JdbcChatMemoryRepository -->
+<dependency>
+    <groupId>org.springframework.ai</groupId>
+    <artifactId>spring-ai-starter-model-chat-memory-repository-jdbc</artifactId>
+</dependency>
```

Comment:
Should we mention the `spring-session-jdbc` one here as well, instead of above, since it's for the persisted approach were it is needed?

Reply:
Or do we also need it for the other since it was changing the session more frequently?

Reply:
spring-session-jdbc is being removed in further commits.